interface/patient_file/summary/add_edit_issue.php

```diff
@@ -647,6 +649,40 @@ function onCodeSelectionChange() {
             document.forms[0].rem_selected_code.disabled = document.forms[0].form_selected_codes.selectedIndex == -1
         }
 
+        function processUdi(udi) {
+            top.restoreSession();
+            let url = '../../../library/ajax/udi.php?udi=' + encodeURIComponent(udi) + '&csrf_token_form=' + <?php echo js_url(CsrfUtils::collectCsrfToken('udi')); ?>;
+            fetch(url, {
+                credentials: 'same-origin',
+                method: 'GET',
+            })
+            .then(response => response.json())
+            .then(data => {
+                let dataJSON = JSON.stringify(data);
+                document.getElementById('udi_data').value = dataJSON;
+                displayUdi(dataJSON);
+            })
+            .catch(error => console.error(error))
+        }
+
+        function displayUdi(dataJSON) {
+            let data = JSON.parse(dataJSON);
+            let display = '<p>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Device")) + ': </span>' + jsText(data.standard_elements.deviceDescription) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Brand Name")) + ': </span>' + jsText(data.standard_elements.brandName) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Company Name")) + ': </span>' + jsText(data.standard_elements.companyName) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Version/Model Number")) + ': </span>' + jsText(data.standard_elements.versionModelNumber) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("DI{{Device Identifier}}")) + ': </span>' + jsText(data.standard_elements.di) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Serial Number")) + ': </span>' + jsText(data.standard_elements.serialNumber) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Lot Number")) + ': </span>' + jsText(data.standard_elements.lotNumber) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Expiration Date")) + ': </span>' + jsText(data.standard_elements.expirationDate) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("Manufacturing Date")) + ': </span>' + jsText(data.standard_elements.manufacturingDate) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("MRI Safety Status")) + ': </span>' + jsText(data.standard_elements.MRISafetyStatus) + '<br>';
+            display += '<span class="font-weight-bold">' + jsText(xl("UDI{{Unique Device Identifier}}")) + ': </span>' + jsText(data.standard_elements.udi) + '<br>';
```

Comment:
You could make a PHP array that maps attribute internal IDs to descriptive names, and then use it to generate the above and the results array in udi.php. Just a thought, not important to change.

Reply:
agreed, gonna likely consolidate this stuff into a class also since need to deal with displaying it in other places in openemr. also may migrate link the udi details into a new sql table since fhir/mu3 requirements appear to require searching of certain items like the expiration date

Reply:
And a MedicalDevice class is born :)